llvm/lib/CheerpUtils/PointerAnalyzer.cpp

```diff
@@ -499,6 +499,12 @@ bool PointerUsageVisitor::visitByteLayoutChain( const Value * p )
 
 PointerKindWrapper& PointerUsageVisitor::visitValue(PointerKindWrapper& ret, const Value* p, bool first)
 {
+	if (p->getType()->isVectorTy())
```

Comment:
I think that there is a better place to do this.
You should add a case to visitRawChain that returns true if the pointee is a vector

Reply:
What does the return type of visitRawChain mean?

Reply:
you pass it a pointer, and it tells you if that pointer is RAW in its context (it refers to linear memory).
I would try to add the vector condition here with an ||:

```
		if (TypeSupport::isAsmJSPointer(p->getType()))
			return true;

```

But it may not be enough, so you can try and see what happens

Reply:
I've removed this check, and instead inserted a check in isInlineableImpl for bitcasts, that always assumed they were on pointers